apps/api/src/app/notification-template/notification-template.controller.ts

```diff
@@ -148,6 +153,50 @@ export class NotificationTemplateController {
     );
   }
 
+  @Get('/blueprint/:templateId')
+  @ApiOkResponse({
+    type: NotificationTemplateResponse,
+  })
+  @ApiOperation({
+    summary: 'Get blueprint notification template',
+  })
+  @ExternalApiAccessible()
+  getNotificationTemplateBlueprintById(
+    @UserSession() user: IJwtPayload,
+    @Param('templateId') templateId: string
+  ): Promise<NotificationTemplateResponse> {
+    return this.getBlueprintNotificationTemplate.execute(
+      GetBlueprintNotificationTemplateCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        templateId,
+      })
+    );
+  }
+
+  @Post('/blueprint/:templateId')
```

Comment:
If the `templateId` param is the notification template ID I would just swap the URLs to:
```
/:templateId/blueprint
```
both for the GET and the POST, as it would be clearer for me that Notification template is the main entity and these are actions for a subresource (blueprint) that is created from a notification template. Also I think it won't clash with future functionality so we would be safe.